tools/export/uvision/__init__.py

```diff
@@ -240,12 +240,10 @@ def generate(self):
                                       self.resources.inc_dirs).encode('utf-8'),
             'device': DeviceUvision(self.target),
         }
-        sct_name, sct_path = self.resources.get_file_refs(
-            FileType.LD_SCRIPT)[0]
-        ctx['linker_script'] = self.toolchain.correct_scatter_shebang(
-            sct_path, dirname(sct_name))
-        if ctx['linker_script'] != sct_path:
-            self.generated_files.append(ctx['linker_script'])
+        sct_file_ref = self.resources.get_file_refs(FileType.LD_SCRIPT)[0]
+        sct_file_ref = self.toolchain.correct_scatter_shebang(sct_file_ref, dirname(sct_file_ref.name))
+        self.resources.add_file_ref(FileType.LD_SCRIPT, sct_file_ref.name, sct_file_ref.path)
```

Comment:
Once we add the new `LD_SCRIPT` ref, don't we also need to remove the existing one? (The one referred to by `self.resources.get_file_refs(FileType.LD_SCRIPT)[0]` above)

Reply:
Why would we need to do that?

Reply:
So we don't have multiple linker scripts in the resources object. I'm assuming that's a bad thing/nonsensical, perhaps my assumption is incorrect?

Reply:
There is nothing wrong with having more than one linker script here, to my knowledge.

Reply:
This is the line that I was worried about: https://github.com/ARMmbed/mbed-os/blob/master/tools/resources/__init__.py#L373

It's only selecting the first option of linker scripts. I suppose in this case, this property isn't being used by the exporter and its directly setting the linker script in the project file to the new one generate, so its not an issue?

Reply:
Yes. That's why it's not an issue

Reply:
Lovely!